ONSdigital/census-rh-ui
# Cr 1813 response error improved logging

# Motivation and Context
We have some errors being logged that are hard to know the cause of since they carry no context other than a message "response error".
This PR adds the originating URL, HTTP method and status.

As well as implementing CR-1813, this also helps understand CR-1822 (AIMS 429 responses).

# What has changed
- use the aiohttp exception to provide extra information in the logging
- improved unit tests

# How to test?
- You would need a failing service such as AIMS or RHSvc to send a response such as a 500 back to see the logged information.
- unit tests simulate this.

# Links
https://collaborate2.ons.gov.uk/jira/browse/CR-1813
